### ISSUE TITLE
Problem with downloading large files.

### ISSUE THREAD

**Issue by NightSommelier** [mgilangjanuar/teledrive#449]
Problem with uploading large files. I downloaded a 7.17 GB file. It downloaded 4 parts up to 1.9 GB in size. When refreshing the folder it shows this information.

<img width="177" src="https://user-images.githubusercontent.com/2665877/230445283-303d3c38-84ce-4172-aebb-c52fde518dda.png">

The file fails to load because the telegram download ended and the file was uploaded by the teledrive server, but the browser thinks that the remaining part of the file, i.e. 14GB, needs to be downloaded

**Comment by greengeckowizard** [comment]
This is common issue try using staging branch since there's a fix implemented. I'll add New update when another issue has been resolved.

**Comment by NightSommelier** [comment]
> This is common issue try using staging branch since there's a fix implemented. I'll add New update when another issue has been resolved.

Can you tell me which one?

**Comment by NightSommelier** [comment]
<img width="455" src="https://user-images.githubusercontent.com/2665877/230447699-ae7726c3-4de2-4ed3-881e-1f9d0e464a92.png">

**Comment by greengeckowizard** [comment]
Staging branch redeploy on staging branch.

**Comment by greengeckowizard** [comment]
What deployment method are you using?

**Comment by NightSommelier** [comment]
Startup problem on the staging branch
<img width="922" alt="image" src="https://user-images.githubusercontent.com/2665877/230473702-213c5e78-f178-4c96-a9c3-a25d2ff33049.png">

**Comment by greengeckowizard** [comment]
Just to make sure are you using yarn from npm?

**Comment by greengeckowizard** [comment]
Could you pull the latest changes and try to run docker-compose up -d.
I edited docker compose file hopefully this resolves the issue.

**Comment by NightSommelier** [comment]
I switched to this commit and everything started.
<img width="922" alt="image" src="https://user-images.githubusercontent.com/2665877/230481047-725092eb-b013-427b-b6f9-2c0ce06d1f26.png">

**Comment by greengeckowizard** [comment]
> I switched to this commit and everything started.
> <img width="922" alt="image" src="https://user-images.githubusercontent.com/2665877/230481047-725092eb-b013-427b-b6f9-2c0ce06d1f26.png">
> 

Awesome let me know if this works for the larger file download.

**Comment by NightSommelier** [comment]
There is a problem, a 2GB file doesn't want to load... it just crashes.

<img width="907" alt="Screenshot 2023-04-06 at 23 03 47" src="https://user-images.githubusercontent.com/2665877/230483569-812068d0-128c-424f-b611-5a9a3daa31b1.png">

**Comment by NightSommelier** [comment]
This file now downloads after the latest fixes. I will check it later with a file that weighs 7 gb
https://github.com/mgilangjanuar/teledrive/issues/449#issuecomment-1499555447

**Comment by greengeckowizard** [comment]
> This file now downloads after the latest fixes. I will check it later with a file that weighs 7 gb [#449 (comment)](https://github.com/mgilangjanuar/teledrive/issues/449#issuecomment-1499555447)

Awesome that's good to hear about it works let me know if that file that's over 7gb downloads correctly and isn't corrupted.

**Comment by NightSommelier** [comment]
> Awesome that's good to hear about it works let me know if that file that's over 7gb downloads correctly and isn't corrupted. 

Unloading and splitting is correct, even split into 4 parts. A little later I will try to download the file and check the integrity.

**Comment by NightSommelier** [comment]
> Unloading and splitting is correct, even split into 4 parts. A little later I will try to download the file and check the integrity.

For some reason it couldn't finish downloading... it doesn't want to continue either.

<img width="377" alt="Screenshot 2023-04-08 at 21 49 38" src="https://user-images.githubusercontent.com/2665877/230738098-4bbe2a39-a9ad-4b81-b98e-bab9ca2f55df.png">
<img width="442" alt="Screenshot 2023-04-08 at 21 50 35" src="https://user-images.githubusercontent.com/2665877/230738103-eb3a585c-d32e-4aad-a5db-2fdf8f6bec4c.png">

**Comment by greengeckowizard** [comment]
I'll make a fix for it thanks for testing.

**Comment by NightSommelier** [comment]
> 15 6084229970/2000000000 (84.81674511299626%)
15 6084595538/2000000000 (84.82184128589412%)
15 6084595538/2000000000 (84.82184128589412%) -end-
GET /api/v1/files/XXXXXXX-4406-43c6-k1vhv5-3a920d32c1a2?raw=1&dl=1&password=null 200 7173382994 - 3089.148 ms
[2023-04-08T18:47:13.529] [INFO] - [Disconnecting from 149.154.167.41:80/TCPFull...]
[2023-04-08T18:47:13.530] [INFO] - [Disconnecting from 149.154.167.41:80/TCPFull complete!]
[2023-04-08T18:47:31.148] [INFO] - [Running gramJS version 2.15.3]
[2023-04-08T18:47:31.151] [INFO] - [Connecting to 149.154.167.41:80/TCPFull using PromisedNetSocket]
[2023-04-08T18:47:31.227] [INFO] - [Connection to 149.154.167.41:80/TCPFull complete!]
[2023-04-08T18:47:31.227] [INFO] - [Using LAYER 150 for initial connect]
undefined
[2023-04-08T18:47:31.470] [INFO] - [Starting direct file download in chunks of 524288 at 0, stride 524288]
[2023-04-08T18:47:31.545] [INFO] - [Connecting to 149.154.167.50:443/TCPFull using PromisedNetSocket]
[2023-04-08T18:47:31.611] [INFO] - [Connection to 149.154.167.50:443/TCPFull complete!]
13 524288/2000000000 (0.007308796985167637%)
GET /api/v1/files/XXXXXXX-4406-43c6-k1vhv5-3a920d32c1a2?raw=1&dl=1&password=null 200 7173382994 - 967.014 ms
13 1048576/2000000000 (0.014617593970335274%) -end-
{ status: 422, body: { error: 'canceled' } }
[2023-04-08T18:47:32.323] [INFO] - [Starting direct file download in chunks of 524288 at 0, stride 524288]
14 1572864/2000000000 (0.021926390955502913%) -end-
{ status: 422, body: { error: 'canceled' } }
[2023-04-08T18:47:33.077] [INFO] - [Disconnecting from 149.154.167.41:80/TCPFull...]
[2023-04-08T18:47:33.078] [INFO] - [Disconnecting from 149.154.167.41:80/TCPFull complete!]

**Comment by NightSommelier** [comment]
I tried to download the file from the last commit. The download was successful, but the file does not correspond to the original, so there is a problem. At some point I get an error to view the file. I showed a screenshot of the size of the two files. On the left side of the file is the original, on the right side is what I was able to download. Here either the file is uploaded to the telegram server is not correct, or the download is collected incorrectly.

I was testing on the f53e28a

<img width="530" alt="Screenshot 2023-04-10 at 14 09 23" src="https://user-images.githubusercontent.com/2665877/230890788-255ab29a-8441-4ded-ae4d-4c00352cd211.png">
